src/lib.rs

```diff
@@ -93,6 +93,7 @@ unsafe fn plrust_call_handler(fcinfo: pg_sys::FunctionCallInfo) -> pg_sys::Datum
         }
         .ok_or(PlRustError::NullFmgrInfo)?
         .fn_oid;
+        unsafe { plrust::recompile_missing_shared_object(fn_oid)?; }
```

Comment:
Shouldn't this happen transparently during evaluation instead of being called in sequence? It repeats a fair amount of the logic inside `evaluate_function`'s Vacant case...? There's a few cases where that can probably be factored out.